src/snowflake/connector/cursor.py

```diff
@@ -1367,13 +1359,13 @@ def reset(self, closing: bool = False) -> None:
         if not self.connection._reuse_results:
             self._result_set = None
 
-    def __iter__(self) -> Iterator[dict] | Iterator[tuple]:
+    def __iter__(self) -> Iterator[dict | None] | Iterator[tuple | None]:
```

Comment:
this might fall out of the scope of this PR but do we consider making the cursor an iterator object (\_\_next__ + __iter__) instead of just an iterable object (\_\_iter__)